tests/test_dataclasses.py

```diff
@@ -489,3 +489,43 @@ class TestClassVar:
 
     tcv = TestClassVar(2)
     assert tcv.klassvar == "I'm a Class variable"
+
+
+def test_unsupported_field_type():
```

Comment:
these shouldn't be in dataclass tests, you might want one test with a dataclass, but in general feature is not specific to dataclasses.

Reply:
Done.